pkg/backend/openshiftcluster.go

```diff
@@ -218,16 +218,19 @@ func (ocb *openShiftClusterBackend) endLease(ctx context.Context, stop func(), d
 	if provisioningState == api.ProvisioningStateFailed {
 		failedProvisioningState = doc.OpenShiftCluster.Properties.ProvisioningState
 	}
-
-	err := ocb.updateAsyncOperation(ctx, doc.AsyncOperationID, doc.OpenShiftCluster, provisioningState, failedProvisioningState)
-	if err != nil {
-		return err
+	// If cluster is in the non-terminal state we are still in the same
+	// operational context and AsyncOperation should not be updated.
+	if provisioningState.IsTerminal() {
+		err := ocb.updateAsyncOperation(ctx, doc.AsyncOperationID, doc.OpenShiftCluster, provisioningState, failedProvisioningState)
+		if err != nil {
+			return err
+		}
 	}
 
 	if stop != nil {
 		stop()
 	}
 
-	_, err = ocb.db.OpenShiftClusters.EndLease(ctx, doc.Key, provisioningState, failedProvisioningState)
+	_, err := ocb.db.OpenShiftClusters.EndLease(ctx, doc.Key, provisioningState, failedProvisioningState)
```

Comment:
It's only possible to redeclare `err` here because there is another value, but since the other value is discarded, there's no need to use `:=`.

Reply:
not true :-)
the previous declaration of err is scoped within the wrapping if statement.

Reply:
Whoops - I was looking at the removed line 222 :persevere: